### PR TITLE
Draft implementations

### PR DESCRIPTION
**Pull request by imfaith-hq**
https://user-images.githubusercontent.com/59166014/135790876-56719304-8a4d-4a0c-9b0a-1293f3061455.mp4

https://user-images.githubusercontent.com/59166014/136009002-8ada1d31-83c4-4d94-b437-385988e50c86.mp4

Implementations made for saving text contained within the textfield when routing off an instance of dm_user_view, channel_page_view and thread_detail_view for whenever we route back to the instance of such particular view.

Implementations made for displaying a list of drafts saved across instances of dm_user_view, channel_page_view and thread_detail_view in the draft UI screen

### CODE REVIEW

**Review by pauleke65 — DISMISSED** [review]
LG

**Comment by imfaith-hq** [comment]
> Nice work. Please make the messages route to where it is being typed.
> 
> There is no point having Draft if I cannot click on the message saved and it takes me to where I typed it.
> 
> So please do that and we are good to go.

Alright sir, but might take me a while, and moreover, unique id of instance of each view for threads and dm's are not yet available.. so i might be at their mercy to get the implementation done i think.

File: lib/ui/view/direct_message/direct_message.dart [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
         viewModelBuilder: () => DirectMessageViewModel(),
         builder: (context, model, child) {
           return Scaffold(
             appBar: AppBar(
               backgroundColor: Colors.white,
               leading: IconButton(
                 icon: const Icon(Icons.arrow_back_ios, color: Colors.black),
-                onPressed: () => Navigator.pop(context),
+                onPressed: () {
+                  return Navigator.pop(context);
```
Comment (jeremiahseun) [inline lib/ui/view/direct_message/direct_message.dart:40]: Replace with model.navigateBack()

File: lib/ui/view/direct_message/direct_message_viewmodel.dart [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+
+  String? get drafts =>
+      _storageService.getString(StorageKeys.currentUserDmDrafts);
+
+
+  void getDraft() async{
+    if(drafts != null){
+      dmDraft = drafts.toString();
+      _storageService.clearData(StorageKeys.currentUserDmDrafts);
+      // print('$draft hhhhh');
```
Comment (jeremiahseun) [inline lib/ui/view/direct_message/direct_message_viewmodel.dart:21]: Remove commented lines

File: lib/ui/view/draft/draft_viewmodel.dart [hunk 3]
```diff
@@ -1,11 +1,139 @@
...
+        duration: const Duration(seconds: 3),
+        variant: SnackbarType.failure,
+        message: errorOccurred,
+      );
+    }
+  }
+
+}
+
+class Objects {
```
Comment (jeremiahseun) [inline lib/ui/view/draft/draft_viewmodel.dart:133]: This class should be on it's own file

**Review by jeremiahseun — DISMISSED** [review]
Please make changes

File: lib/ui/view/draft/draft_viewmodel.dart (hunk 3 above)
Comment (imfaith-hq) [reply lib/ui/view/draft/draft_viewmodel.dart]: done

**Review by pauleke65 — APPROVED** [review]
LG

**Review by YenumJay — APPROVED** [review]

**Pull request merged**